components/MemoryGrid/index.js

```diff
@@ -60,12 +64,48 @@ export default function MemoryGrid() {
     );
   }, []);
 
-  const handleClick = (event) => {
+  useEffect(() => {
+    if (compareImages.length === 3) {
```

Comment:
Why do you need a `useEffect()` for implementing this logic? Shouldn't all of this happen on `handleClick`?
It works but it causes a rerender which isn't necessary from my point of view.
It's not a blocker for this PR, but maybe you can consider this in your next refactoring.

Reply:
The `useEffect()` is necessary here, because when i put all of it in the `handleClick` the states get updated one turn delayed.

Reply:
Or is there another way I can fix the problem?